src/transformers/models/bloom/modeling_flax_bloom.py

```diff
@@ -379,25 +383,18 @@ def __call__(
         attn_weights = attn_weights.reshape(output_size)
 
         # TODO: apply softmax to attention weights
-        att_probs = self.scale_mask_softmax(attn_weights, attention_mask, causal_mask, layer_number)
+        attention_probs = self.scale_mask_softmax(attn_weights, attention_mask, causal_mask, layer_number)
+
         if head_mask is not None:
             attention_probs = attention_probs * head_mask
 
         output_size = (value.shape[0], value.shape[2], query.shape[1], value.shape[3])
 
-        value = jnp.transpose(value, (1, 0, 2, 3)).reshape(value.shape[1], output_size[0] * output_size[1], -1)
-        attention_probs_reshaped = jnp.reshape(att_probs, (output_size[0] * output_size[1], output_size[2], -1))
-
-        context = jnp.matmul(attention_probs_reshaped, jnp.transpose(value, (1, 0, 2)))
-        context = context.reshape(output_size)
-        context = jnp.transpose(context, (2, 0, 1, 3))
-
-        # [q_length, batch_size, num_heads, head_dim] --> [q_length, batch_size, hidden_size]
-        new_context_layer_shape = context.shape[:-2] + (self.hidden_size,)
-        context = context.reshape(new_context_layer_shape)
+        attn_output = jnp.einsum("...hqk,...khd->...qhd", attention_probs, value)
```

Comment:
@younesbelkada let's try to keep it simple - no need for all this manual reshaping. Our existing Flax models are quite optimized and we should follow the pattern that is already used in other models. So instead of copying from BLOOM PyTorch we should copy from OPT Flax

Reply:
Resolved in https://github.com/huggingface/transformers/pull/18022/commits/3444985451749540bd862910ebe7d01da8413771